src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,171 @@ def test_set_beam_shift(self):
+        """
+        detector = 'se-detector'  # autostigmator only works for the se-detector with channel=electron1
+        channel = DETECTOR2CHANNELNAME[detector]
+
+        # Start auto stigmation and check if it is running.
+        autostigmator_future = self.scanner.applyAutoStigmator(detector)
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(channel)
+        self.assertEqual(autostigmator_state, True)
+        self.assertIsInstance(autostigmator_future, ProgressiveFuture)
+
+        # Stop auto stigmation and check if it stopped running.
+        autostigmator_future.cancel()
+        time.sleep(5.0)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(channel)
+        self.assertEqual(autostigmator_state, False)
+
+        # Create lists of stigmator values to check if the results are reproducible.
+        stigmator_values_x = []  # List of stigmator values in x direction
+        stigmator_values_y = []  # List of stigmator values in y direction
+
+        max_execution_time = 30  # Approximately 3 times the normal expected execution time (s)
+        for i in range(0, 5):
+            starting_time = time.time()
+            autostigmator_future = self.scanner.applyAutoStigmator(detector)
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            autostigmator_state = self.microscope.is_autostigmating(channel)
+            self.assertEqual(autostigmator_state, True)
+            self.assertIsInstance(autostigmator_future, ProgressiveFuture)
+
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            autostigmator_future.result(timeout=max_execution_time)
+
+            # Check if the time to perform the auto stigmation is not too long
+            self.assertLess(time.time() - starting_time, 30,
+                            "Execution of auto stigmation was stopped because it took more than %s seconds." %
+                            max_execution_time)
+            # Line to inspect the execution time to update the expected time
+            print("Execution time was %s seconds " % (time.time() - starting_time))
+
+            stigmator_values_x.append(self.microscope.get_stigmator()[0])
+            stigmator_values_y.append(self.microscope.get_stigmator()[1])
+
+        # Check if all the stigmator values are within the required range each run
+        stigmator_range = self.microscope.stigmator_info()['range']
+        min_x_stig_values, max_x_stig_values = stigmator_range['x']
+        min_y_stig_values, max_y_stig_values = stigmator_range['y']
+        # TODO When auto stigmation works implement a check on the reproducibility of the auto stigmation values and
+        #  define the range with which they may differ from each other. In theory similar values should be returned.
```

Comment:
No need for that. When auto stigmation works, this driver will allow to call it. No need to add tests to check that it works well and reliably, because if it doesn't, there is no change in Odemis that will fix it anyway.